pkg/partition/partition.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package partition
+
+import (
+	"fmt"
+
+	"github.com/diskfs/go-diskfs"
+	"github.com/diskfs/go-diskfs/partition/gpt"
+	"k8s.io/klog"
+)
+
+const (
+	// BytesRequiredForGPTHeader is the total bytes required to store the GPT header
+	// 128 bytes are required per partition, total no of partition supported by GPT = 128
+	// Therefore, total bytes = 128*128
+	BytesRequiredForGPTHeader = 16384
+
+	// GPTPartitionStartByte is the byte on the disk at which the first partition starts.
+	// Normally partition starts at 1MiB, (as done by fdisk utility). This is done to
+	// align the partition start to physical block sizes on the disk.
+	GPTPartitionStartByte = 1048576
+)
+
+// Disk struct represents a disk which needs to be partitioned
+type Disk struct {
+	// DevPath is the /dev/sdX entry of the disk
+	DevPath string
+	// DiskSize is size of disk in bytes
+	DiskSize uint64
+	// LogicalBlockSize is the block size of the disk normally 512 or 4k
+	LogicalBlockSize uint64
+
+	table *gpt.Table
+}
+
+// ApplyPartitionTable applies the partition table to the given disk.
+func (d *Disk) ApplyPartitionTable() error {
+	if len(d.table.Partitions) == 0 {
+		return fmt.Errorf("no partitions specified in partition table")
+	}
+	diskFD, err := diskfs.Open(d.DevPath)
```

Comment:
Do we need to close the disk opened here?

Reply:
The diskfs library does not have a close() method. I will check it and update in the upstream repo.